pkg/tsdb/cloudwatch/request_parser.go

```diff
@@ -13,14 +13,16 @@ import (
 	"github.com/google/uuid"
 	"github.com/grafana/grafana-plugin-sdk-go/backend"
 	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
 )
 
 var validMetricDataID = regexp.MustCompile(`^[a-z][a-zA-Z0-9_]*$`)
 
 // parseQueries parses the json queries and returns a map of cloudWatchQueries by region. The cloudWatchQuery has a 1 to 1 mapping to a query editor row
 func (e *cloudWatchExecutor) parseQueries(queries []backend.DataQuery, startTime time.Time, endTime time.Time) (map[string][]*cloudWatchQuery, error) {
 	requestQueries := make(map[string][]*cloudWatchQuery)
-	migratedQueries, err := migrateLegacyQuery(queries, startTime, endTime)
+
+	migratedQueries, err := migrateLegacyQuery(queries, e.features.IsEnabled(featuremgmt.FlagCloudWatchDynamicLabels), startTime, endTime)
```

Comment:
Is there a reason for passing this feature toggle as an argument to migrateLegacyQuery? If it's for testing purposes, there's other ways to mock it.

Reply:
It is not just for testing.
Hmm I don't think there is a way to avoid passing that information to `migrateLegacyQuery` from the `executor` unless we eliminate this function call altogether.

Reply:
My bad - didn't see it was a method. This is fine :+1:

Reply:
minor thing but you could make `migrateLegacyQuery` a method of `cloudWatchExecutor` so you can access the properties there